tests/test_analysis.py

```diff
@@ -0,0 +1,87 @@
+import covasim as cv
+import unittest
+import numpy as np
+
+
+# Suite of tests to test basic functionality of the analysis.py file
+def test_analysis_snapshot():
+    sim = cv.Sim(analyzers=cv.snapshot('2020-04-04', '2020-04-14'))
+    sim.run()
+    snapshot = sim['analyzers'][0]
+    people = snapshot.snapshots[0]             # Option 1
+    people2 = snapshot.snapshots['2020-04-04'] # Option 2
+    people3 = snapshot.get('2020-04-14')       # Option 3
+    people4 = snapshot.get(34)                 # Option 4
+    people5 = snapshot.get()                   # Option 5
+    # people3 = []  # uncomment to verify error
+    i = 0
+    peoples = [people, people2, people3, people4, people5]
+    for p in peoples:
+        i+=1
+        if len(p) == 0:
+            raise ValueError("Option {} is not getting people array correctly".format(i))
+
+
+def test_analysis_hist():
+    # raising multiple histograms to check windows functionality
+    sim = cv.Sim(analyzers=cv.age_histogram(days=["2020-03-30", "2020-03-31", "2020-04-01"]))
+    sim.run()
+
+    # checks to make sure dictionary form has right keys
+    agehistDict = sim['analyzers'][0].get()
+    assert len(agehistDict.keys()) == 5
+
+    # checks to see that compute windows is correct
+    agehist = sim['analyzers'][0]
+    try:
+        agehist.compute_windows()
+    except ValueError:
+        raise ValueError("Unable to compute windows")
+
+    # checks compute_windows and plot()
+    try:
+        agehist.plot(windows=True)
+    except:
+        raise ValueError("Cannot plot this histogram with windows")
+
+
+def test_analysis_fit():
+    sim = cv.Sim(datafile="example_data.csv")
+    sim.run()
+    fit = sim.compute_fit()
+    # battery of tests to test basic fit function functionality
+    # tests that running functions does not produce error
+    try:
+        fit.compute_losses()
+    except:
+        raise ValueError("Unable to compute losses")

```

Comment:
I think taking out the try except blocks here would be useful.  If an exception is raised out of the fit.compute_losses() method, let's let that call stack show up in the output. This pattern might end up swallowing that error and all we see is the value error.

Reply:
Down the road, it would be interesting to see for each of these some kind of state change that should be expected by the method under test.  So if the "fit" object is different in some way after calling the compute losses method, let's make sure of that. Minimal example here:
```
from copy import deepcopy # 
my_secret_property_start = deepcopy(fit.some_changing_value)
fit.compute_losses()
my_secret_property_end = deepcopy(fit.some_changing_value)
assert my_secret_property_start != my_secret_property_end, msg="Expected that something should change when computing losses of fit"
```

But getting this level of testing should _not_ block getting this PR merged.

Reply:
Agree. If you really want to use this, then I think better practice would be something like:
```
    except Exception as E:
        raise ValueError(f"Unable to compute losses: {E}")
```
But this should only be done in cases where it's necessary to add extra information:
https://github.com/InstituteforDiseaseModeling/covasim/blob/master/covasim/misc.py#L308
But actually the more modern way of doing this is using "raise ... from":
https://github.com/InstituteforDiseaseModeling/covasim/blob/master/covasim/sim.py#L211

Reply:
Re: change of state, that's a little tricky. I think it might be a little brittle to test what in a non-Python context might be private attributes. Here I think it would be better to just test the expected output, e.g. if data and sim are the same, the loss is zero, if they're not the same, it's nonzero. It shouldn't matter what's happening with secret properties as long as that condition holds!